todo_rxbloc_app/lib/feature_todos_list/bloc/todos_list_manage_bloc.dart

```diff
@@ -24,26 +28,51 @@ abstract class TodosListManageBlocStates {
 @RxBloc()
 class TodosListManageBloc extends $TodosListManageBloc {
   final TodoService todoService;
-
-  TodosListManageBloc({required this.todoService}) {
+  final CoordinatorBlocType coordinatorBloc;
+  final NavigationBlocType navigationBloc;
+  TodosListManageBloc(
+      {required this.navigationBloc,
+      required this.coordinatorBloc,
+      required this.todoService}) {
     todoUpdated.connect().addTo(_compositeSubscription);
     todoAdded.connect().addTo(_compositeSubscription);
+    coordinatorBloc.states.todoDeleted
+        .bind(_deleteSubject)
+        .addTo(_compositeSubscription);
+    coordinatorBloc.states.todoUpdated
+        .bind(_updateSubject)
+        .addTo(_compositeSubscription);
   }
+
+  final _deleteSubject = BehaviorSubject<TodoEntity>();
+  final _updateSubject = BehaviorSubject<TodoEntity>();
+
   @override
   ConnectableStream<void> _mapToTodoAddedState() =>
       _$addTodoEvent.doOnData((todo) {
         todoService.addTodo(todo);
       }).publish();
 
   @override
-  Stream<TodoEntity> _mapToTodoDeletedState() =>
-      _$deleteTodoEvent.doOnData((todo) {
-        todoService.delteTodo(todo);
-      });
+  Stream<TodoEntity> _mapToTodoDeletedState() => Rx.merge([
+        _$deleteTodoEvent.doOnData((todo) {
+          todoService.delteTodo(todo);
+        }),
+        _deleteSubject.doOnData((todo) {
+          todoService.delteTodo(todo);
+          navigationBloc.events.navigate(
+            const NavigationParams(navigationEnum: NavigationEnum.todosList),
+          );
+        })
+      ]);
 
   @override
-  ConnectableStream<void> _mapToTodoUpdatedState() =>
-      _$updateTodoEvent.doOnData((todo) {
-        todoService.updateTodo(todo.copyWith(complete: !todo.complete));
-      }).publish();
+  ConnectableStream<void> _mapToTodoUpdatedState() => Rx.merge([
+        _$updateTodoEvent.doOnData((todo) {
```

Comment:
doOnDate is usually used for handling side-effects. In your case updating a todo is the main action you want to perform. So please explore operators such as switchMap, flatMap etc. Also you can check [this](https://medium.com/prime-holding-jsc/introducing-rx-bloc-part-2-faf956f2bd99#c627) article to learn how to implement State Handlers